pkg/runner/healthcheck_common.go

```diff
@@ -0,0 +1,256 @@
+			hh.report.Fixes[i] = fixhc
+			return nil
+		})
+	}
+	return eg.Wait()
+}
+
+// DefaultContainerChecker returns a Checker, a method which when executed will check for the
+// existance of the container. This should be considered a sensible default for checking whether
+// docker containers are started.
+func DefaultContainerChecker(ctx context.Context, log *zap.SugaredLogger, cli *client.Client, name string) Checker {
+	return func() bool {
+		ci, err := docker.CheckContainer(ctx, log, cli, name)
+		if err != nil || ci == nil {
+			return false
+		}
+		return ci.State.Running
+
+	}
+}
+
+// DefaultContainerFixer returns a Fixer, a method which when executed will ensure the named
+// container exists with some default paramaters which are appropriate for infra containers.
+// Unless containers require special consideration, this should be considered the sensible default
+// fixer for docker containers.
+func DefaultContainerFixer(ctx context.Context, log *zap.SugaredLogger, cli *client.Client, containerName string, imageName string, networkID string, portSpecs []string, pull bool, customHostConfig *container.HostConfig, cmds ...string) Fixer {
+	// Docker host configuration.
+	// https://godoc.org/github.com/docker/docker/api/types/container#HostConfig
+	var hostConfig container.HostConfig
+	if reflect.DeepEqual(*customHostConfig, container.HostConfig{}) {
+		hostConfig = container.HostConfig{
+			NetworkMode: container.NetworkMode(networkID),
+			Resources: container.Resources{
+				Ulimits: []*units.Ulimit{
+					{Name: "nofile", Hard: InfraMaxFilesUlimit, Soft: InfraMaxFilesUlimit},
+				},
+			},
+		}
+	} else {
+		hostConfig = *customHostConfig
+	}
+	// Try to parse the portSpecs, but if we can't, fall back to using random host port assignments.
+	// the portSpec should be in the format ip:public:private/proto
+	_, portBindings, err := nat.ParsePortSpecs(portSpecs)
+	if err != nil {
+		hostConfig.PublishAllPorts = true
+	} else {
+		hostConfig.PortBindings = portBindings
+	}
+
+	// Configuration for the container:
+	containerConfig := container.Config{
+		Image: imageName,
+		Cmd:   cmds,
+	}
+
+	ensure := docker.EnsureContainerOpts{
+		ContainerName:      containerName,
+		ContainerConfig:    &containerConfig,
+		HostConfig:         &hostConfig,
+		PullImageIfMissing: pull,
+	}
+
+	// Make sure this container is running when the closure is executed.
+	return func() error {
+		_, _, err := docker.EnsureContainer(ctx, log, cli, &ensure)
+		return err
+	}
+}
+
+// CustomContainerFixer returns a Fixer, a method which when executed will ensure the named
+// container exists. Unlike the DefaultContainerFixer, a custom image may be built for the
+// container.
+func CustomContainerFixer(ctx context.Context, log *zap.SugaredLogger, cli *client.Client, buildCtx string, containerName string, imageName string, networkID string, portSpecs []string, pull bool, customHostConfig *container.HostConfig, cmds ...string) Fixer {
```

Comment:
Such long argument lists can get difficult to work with. Can we encapsulate these in a struct, please? Same for `DefaultContainerFixer`.

Reply:
Sure. This definitely grew more complex as I started adding to it. Can swap this out with a struct.